crates/modelardb_common/src/storage.rs

```diff
@@ -68,6 +69,10 @@ impl DeltaLake {
     /// Create a new [`DeltaLake`] that manages the delta tables in `data_folder_path` if it exists.
     /// Returns a [`DeltaTableError`] if `data_folder_path` is not accessible.
```

Comment:
Maybe change this to say that an error is returned if the path cannot be created if it does not already exist.  I am not sure `LocalFileSystem::new_with_prefix` can fail now that we create the path before.